### PR TITLE
Removing Globals Part 2 of 4 (probably)

### PR DESCRIPTION
**Pull request by Neelesh99**
# Summary

The Variable map has been removed from global scope, now encapsulated in the game state under the RuleContext struct. Should help with fixing WASM NaN outputs.

## Additional Information

I've had to adjust one or two client implementations to suit the new non-global maps, pls let me know if I've done it wrong.

## Test Plan

Adjusted a bunch of tests to fit with the no-longer-global VariableMap

P.S. Both us (Team 3) and Team 6 (@Regnigh) didn't instantiate our baseRoles when loading up the custom role implementations. I've instantiated them in this PR, we'll have to be wary of this in the future. (Thanks @lhl2617 for the help finding this issue)

### CODE REVIEW

File: internal/clients/team2/client.go [hunk 1]
```diff
@@ -184,7 +184,7 @@ func (c *client) gameConfig() config.ClientConfig {
 // You will need it to access the game state through its GetGameStateMethod.
 func (c *client) Initialise(serverReadHandle baseclient.ServerReadHandle) {
 	c.ServerReadHandle = serverReadHandle
-	c.LocalVariableCache = rules.CopyVariableMap()
+	c.LocalVariableCache = rules.CopyVariableMap(c.ServerReadHandle.GetGameState().RulesInfo.VariableMap)
```
Comment (NGovani) [inline internal/clients/team2/client.go:187]: make sure to confirm with team 2 when changing this code

**Review by NGovani — APPROVED** [review]
just check with the teams whose code your changing, also linter issues otherwise LGTM.
Pre approving would be good for LH to check it aswell

File: internal/clients/team2/client.go (hunk 1 above)
Comment (Eirikalb) [reply internal/clients/team2/client.go:187]: Team 2 approves

Comment (Neelesh99) [reply internal/clients/team2/client.go:187]: Cheers

**Pull request merged**